frontend/server/src/Controllers/Contest.php

```diff
@@ -438,6 +442,53 @@ public static function getContestDetailsForSmarty(
         return $result;
     }
 
+    /**
+     * @return array{payload: array{status: string, contests: array{title: string, description: string, start_time: int, finish_time: int, last_updated: int, window_length: int, rerun_id: int, admission_mode: string, alias: string, scoreboard: int, points_decay_factor: float, partial_score: int, submissions_gap: int, feedback: string, penalty: int, penalty_type: string, penalty_calc_policy: string, show_scoreboard_after: int, urgent: int, languages: string, recommended: int, scoreboard_url: string, scoreboard_url_admin: string}[]}, privateContestsAlert: bool}
+     */
+    public static function getContestListMineForSmarty(
+        \OmegaUp\Request $r
+    ) : array {
+        // Authenticate user
+        $r->ensureMainUserIdentity();
+
+        // If the user have private material (contests/problems), an alert is issued
+        // suggesting to contribute to the community by releasing the material to
+        // the public. This flag ensures that this alert is shown only once per
+        // session, the first time the user visits the "My contests" page.
+        $privateContestsAlert = (
+            !isset($_SESSION['private_contests_alert']) &&
+            \OmegaUp\DAO\Contests::getPrivateContestsCount($r->user) > 0
+        );
+
+        if ($privateContestsAlert) {
+            $_SESSION['private_contests_alert'] = true;
+        }
+
+        return [
+            'payload' => self::getContestListInternal(
+                $r,
+                '\OmegaUp\DAO\Contests::getAllContestsOwnedByUser'
+            ),
+            'privateContestsAlert' => $privateContestsAlert,
+        ];
+    }
+
+    /**
+     * @return array{LANGUAGES: array<int, string>, IS_UPDATE?: int}
```

Comment:
```suggestion
     * @return array{LANGUAGES: string[], IS_UPDATE?: int}
```

recuerda que `T[]` casi siempre es mejor que `array<int, T>`.

Reply:
Listo